### ISSUE TITLE
TextBoxComponent exception 

### ISSUE THREAD

**Issue by unger1984** [flame-engine/flame#532]
## Description

Example for [text](https://github.com/flame-engine/flame/blob/a3fa58377681cb3505bdb51e5af6729cdf4b8f77/doc/examples/text/lib/main.dart) in documentation not work.

StackTrace:

```
E/flutter (32743): Tried calling: x
E/flutter (32743): #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
E/flutter (32743): #1      Vector2Extension.toRect (package:flame/extensions/vector2.dart:23:40)
E/flutter (32743): #2      TextBoxComponent._redrawCache (package:flame/components/text_box_component.dart:151:48)
E/flutter (32743): #3      TextBoxComponent.redrawLater (package:flame/components/text_box_component.dart:179:20)
E/flutter (32743): #4      new TextBoxComponent (package:flame/components/text_box_component.dart:77:5)
E/flutter (32743): #5      new MyTextBox (package:abaron/main.dart:23:9)
E/flutter (32743): #6      new MyGame (package:abaron/main.dart:53:9)
E/flutter (32743): #7      main (package:abaron/main.dart:15:10)
```

## Development environment

Flame version:

1.0.0-rc2

Flutter doctor:

```
 Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, 1.23.0-18.1.pre, on Mac OS X 10.15.7 19H15 x86_64, locale ru)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 12.1.1)
[!] Android Studio (not installed)
[✓] IntelliJ IDEA Ultimate Edition (version 2020.2.3)
[✓] VS Code (version 1.51.1)
[✓] Connected device (1 available)
```

## Runtime

This issue is related to running on which platform? (Select all that apply)

 - [x] Android :android:

## Minimal reproducible code (Required for bugs)

Cropped example code:

```dart
import 'dart:ui';

import 'package:flame/anchor.dart';
import 'package:flame/components/text_box_component.dart';
import 'package:flame/extensions/vector2.dart';
import 'package:flame/flame.dart';
import 'package:flame/game.dart';
import 'package:flame/palette.dart';
import 'package:flame/text_config.dart';
import 'package:flutter/material.dart';

void main() async {
  final Vector2 size = await Flame.util.initialDimensions();
  runApp(MyGame(size).widget);
}

TextConfig regular = TextConfig(color: BasicPalette.white.color);
TextConfig tiny = regular.withFontSize(12.0);

class MyTextBox extends TextBoxComponent {
  MyTextBox(String text)
      : super(text, config: tiny, boxConfig: TextBoxConfig(timePerChar: 0.05));

  @override
  void drawBackground(Canvas c) {
    final Rect rect = Rect.fromLTWH(0, 0, width, height);
    c.drawRect(rect, Paint()..color = const Color(0xFFFF00FF));
    c.drawRect(
        rect.deflate(boxConfig.margin),
        Paint()
          ..color = BasicPalette.black.color
          ..style = PaintingStyle.stroke);
  }
}

class MyGame extends BaseGame {
  MyGame(Vector2 screenSize) {
    size = screenSize;

    add(MyTextBox(
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eget ligula eu lectus lobortis condimentum.',
    )
      ..anchor = Anchor.bottomLeft
      ..y = size.y);
  }
}
```

**Comment by spydon** [comment]
This is indeed a bug, the TextBoxComponent seems to have some missing pieces to work with v1.0.0, we are looking into it.
Thanks for reporting it!

**Comment by spydon** [comment]
@unger1984 fixed it here #534. 
It will probably be merged in a few days, but it might take some time until we release a new RC so you might want to depend directly on that branch for a while if you are in a hurry.

**Comment by unger1984** [comment]
@spydon Thank you for your reply and for making such a great tool!
